python/ray/runtime_env/ARCHITECTURE.md

```diff
@@ -0,0 +1,66 @@
+# Runtime Env Architecture
+
+This document describes the architecture of Ray's runtime environment feature.
+
+## Overview
+
+For a high-level overview of runtime environments, see the blog post [here](https://www.anyscale.com/blog/handling-files-and-packages-on-your-cluster-with-ray-runtime-environments).  The blog post also contains a simple example of how to use runtime environments.
+
+For a more detailed explanation of how to use runtime environments, see the [documentation](https://docs.ray.io/en/latest/ray-core/handling-dependencies.html).
+
+## Architecture
```

Comment:
suggest to add info about the runtime_env agent fate sharing with the raylet (and why that's important)

Reply:
Added